luaa.c

```diff
@@ -150,13 +154,56 @@ extern const struct luaL_Reg awesome_mouse_meta[];
  * @signal refresh
  */
 
-/** Awesome is about to enter the event loop.
+/** AwesomeWM is about to enter the event loop.
  *
  * This means all initialization has been done.
  * @signal startup
  */
 
-/** Awesome is exiting / about to restart.
+/** AwesomeWM is about to scan for existing screens.
+ *
+ * Connect to this signal when code needs to be executed after the lua context
+ * is initialized and modules loaded, but before screens are added.
+ *
+ * To manage screens manually, set `screen.automatic_factory = false` and
+ * connect to the `area::detected` signal. It is then possible to use
+ * `screen.fake_add` to create virtual screens. Be careful when using this,
+ * when done incorrectly, no screens will be created. Using Awesome with zero
+ * screen is **not** supported.
+ *
+ * @signal scanning::screen
+ * @see area::detected
+ * @see screen.fake_add
+ */
+
+/** AwesomeWM is about to scan for existing clients.
+ *
+ * Connect to this signal when code needs to be executed after screens are
+ * initialized, but before clients are added.
+ *
+ * @signal scanning::client
+ */
+
+/** AwesomeWM is done scanning for screens.
+ *
+ * Connect to this signal to execute code after the screens have been created,
+ * but before the clients are added. This signal can also be used to split
+ * physical screens into multiple virtual screens before the clients (and their
+ * rules) are executed.
+ *
+ * @signal scanned::screen
```

Comment:
The difference is literally that they are different signals. There is no way to re-order signals connections. So, given modules needs to assume they are loaded in random order, having more signals allows for finer control over when the code is executed.

There's also the case that `scanned::screen` wont be emitted in `auto` mode because `rc.lua` is only initialized *after* they should have been sent. This makes it easier for module specifically designed to only work in `manual` mode to have less side effects when accidentally loaded in `auto` mode.

Also, because: consistency ;)